qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -0,0 +1,550 @@
+from qiskit_experiments.data_processing.processor_library import get_processor
+from qiskit_experiments.data_processing.exceptions import DataProcessorError
+from qiskit_experiments.exceptions import AnalysisError
+
+from .curve_data import CurveData, SeriesDef, FitData, ParameterRepr, FitOptions
+from .data_processing import multi_mean_xy_data, data_sort
+from .curve_fit import multi_curve_fit
+from .visualization import MplCurveDrawer, BaseCurveDrawer
+
+PARAMS_ENTRY_PREFIX = "@Parameters_"
+DATA_ENTRY_PREFIX = "@Data_"
+
+
+class BaseCurveAnalysis(BaseAnalysis, ABC):
+    """Abstract superclass of curve analysis base classes.
+
+    Note that this class doesn't define :meth:`_run_analysis` method,
+    and no actual fitting protocol is implemented in this base class.
+    However, this class defines several common methods that can be reused.
+    A curve analysis subclass can construct proper fitting protocol
+    by combining following methods, i.e. subroutines.
+    See :ref:`curve_analysis_workflow` for how these subroutines are called.
+
+    .. rubric:: _generate_fit_guesses
+
+    An abstract method to create initial guesses for the fit parameters.
+    This should be implemented by subclass.
+    See :ref:`curve_analysis_init_guess` for details.
+
+    .. rubric:: _format_data
+
+    A method to format curve data. By default, this method takes the average of y values
+    over the same x values and then sort the entire data by x values.
+
+    .. rubric:: _evaluate_quality
+
+    A method to evaluate the quality of the fit based on the fit result.
+    This returns "good" when reduced chi-squared is less than 3.0.
+    Usually it returns string "good" or "bad" according to the evaluation.
+    This criterion can be updated by subclass.
+
+    .. rubric:: _run_data_processing
+
+    A method to perform data processing, i.e. create data arrays from
+    a list of experiment data payload.
+
+    .. rubric:: _run_curve_fit
+
+    A method to perform fitting with predefined fit models and formatted data.
+    This method internally calls :meth:`_generate_fit_guesses`.
+    Note that this is a core functionality of the :meth:`_run_analysis` method,
+    that creates fit result object from the processed curve data.
+
+    .. rubric:: _create_analysis_results
+
+    A method to create analysis results for important fit parameters
+    that might be defined by analysis options ``result_parameters``.
+    In addition, another entry for all fit parameters is created when
+    the analysis option ``return_fit_parameters`` is ``True``.
+
+    .. rubric:: _create_curve_data
+
+    A method to create analysis results for data points used for the fitting.
+    Entries are created when the analysis option ``return_data_points`` is ``True``.
+    If analysis consists of multiple series, analysis result is created for
+    each curve data in the series.
+
+    .. rubric:: _preparation
+
+    A method that should be called before other methods are called.
```

Comment:
This reads like it is something a developer needs to do themselves, if it is already handled in the base class it should say something
```suggestion
    An initialize method that is called before other methods are called.
```